src/routers/handlers/upload/upload.ts

```diff
@@ -62,13 +64,13 @@ export class UploadHandler extends GenericHandler {
             return undefined;
         }
     }
-}
-
 
-export function getFileUploadUrl(req: Request): string{
-    const zipPortalBaseURL = `${req.protocol}://${req.get('host')}`;
-    const zipPortalCallbackUrl = encodeURIComponent(`${zipPortalBaseURL}${PrefixedUrls.UPLOADED}/${fileIdPlaceholder}`);
-    const xbrlValidatorBackUrl = encodeURIComponent(zipPortalBaseURL + PrefixedUrls.HOME);
+    private getRedirectUrl(req: Request, companyNumber: string): string {
+        const base = RedirectUrl.getUriBase(req);
+        const zipPortalCallbackUrl = encodeURIComponent(`${base}${PrefixedUrls.COMPANY_AUTH_UPLOAD}`);
```

Comment:
This isn't populating the company number. This leaves the zipPortalCallbackUrl incorrect. e.g. `'http://chs.local/xbrl_validate/submit?callback=http://chs.local/accounts-filing/company/:companyNumber/upload&backUrl=http://chs.local/accounts-filing/confirm-company?companyNumber=00006400'` has the :companyNumber placeholder. 


This can be fixed by populating that parameter, but I would recommend removing the `/company/:companyNumber` all together. We don't need it as the companyNumber is stored in the session so we don't need it as a path parameter.